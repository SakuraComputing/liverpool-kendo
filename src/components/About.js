import React from 'react';

const About = () =>  (
  <div className="content-container">
      <h1>General Definition</h1>
      Kendo (剣道 kendō, lit. "sword way") is a modern Japanese martial art, which descended from swordsmanship (kenjutsu) and uses bamboo swords (shinai) and protective armour (bōgu). Today, it is widely practiced within Japan and many other nations across the world.

      Kendo is an activity that combines martial arts practices and values with strenuous sport-like physical activity.

      <h1>All Japan Kendo Federation defintion of Kendo</h1>
      <div>
          <h2>The Concept of Kendo</h2>

          The concept of Kendo is to discipline the human character through the application of the principles of the Katana (sword).

          <h2>The Purpose of Practicing Kendo</h2>

          The purpose of practicing Kendo is:

          To mold the mind and body,

          To cultivate a vigorous spirit,

          And through correct and rigid training,

          To strive for improvement in the art of Kendo,

          To hold in esteem human courtesy and honor,

          To associate with others with sincerity,

          And to forever pursue the cultivation of oneself.

          This will make one be able:

          To love his/her country and society,

          To contribute to the development of culture

          And to promote peace and prosperity among all peoples.

          (The Concept of Kendo was established by All Japan Kendo Federation in 1975.)


          <h2>“The Mindset of Kendo Instruction” and its explanation</h2>

          <h3>The Significance of the Shinai</h3>

          For the correct transmission and development of Kendo, efforts should be made to teach the correct way of handling the shinai in accordance with the principles of the sword.

          Kendo is a way where the individual cultivates one’s mind (the self) by aiming for shin-ki-ryoku-itchi (unification of mind, spirit and technique) utilizing the shinai. The “shinai-sword” should be not only directed at one’s opponent but also at the self. Thus, the primary aim of instruction is to encourage the unification of mind, body and shinai through training in this discipline.

          <h3>Reiho - Etiquette</h3>

          When instructing, emphasis should be placed on etiquette to encourage respect for partners, and nurture people with a dignified and humane character.

          Even in competitive matches, importance is placed on upholding etiquette in Kendo. The primary emphasis should thus be placed on instruction in the spirit and forms of reiho (etiquette) so that the practitioner can develop a modest attitude to life, and realize the ideal of koken-chiai (the desire to achieve mutual understanding and betterment of humanity through Kendo.)

          <h3>Lifelong Kendo</h3>

          While providing instruction, students should be encouraged to apply the full measure of care to issues of safety and health, and to devote themselves to the development of their character throughout their lives.

          Kendo is a “way of life” that successive generations can learn together. The prime objective of instructing Kendo is to encourage the practitioner to discover and define their way in life through training in the techniques of Kendo. Thus, the practitioner will be able to develop a rich outlook on life and be able to put the culture of Kendo into use, thereby benefitting from its value in their daily lives through increased social vigour.

          <p></p>
          March 14, 2007
          All Japan Kendo Federation

          * Translation of ‘Kendo Shido no Kokoro Gamae (The Mindset of Kendo Instruction)’ and its explanation in Japanese

      </div>

  </div>
);
export default About;
