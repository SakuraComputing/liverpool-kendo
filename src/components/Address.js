import React from 'react';
import SimpleMap from './GoogleMap';

export const Address = () => {
    return (
        <div className="content-container">
            <p></p>
            <li>The Egg Dance Studio</li>
            <li>IM Marsh Campus</li>
            <li>Mossley Hill Road</li>
            <li>Aigburth</li>
            <li>Garston</li>
            <li>L17 6BD</li>

            <li>Wednesdays 7.30pm to 9.30pm</li>

            <li>Contact: 07752 969106</li>

            <div style={{width: '100%', height: '400px'}}>
                <SimpleMap/>
            </div>


        </div>

    );
};
export default Address;
