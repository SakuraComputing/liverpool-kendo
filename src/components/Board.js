import React from 'react'
import Note from './Note'

class Board extends React.Component {

    static defaultProps = {
        count: 10
    };

    constructor() {
        super();

        this.state = {
            notes: []
        };

        this.nextId = this.nextId.bind(this);
        this.add = this.add.bind(this);
        this.update = this.update.bind(this);
        this.remove = this.remove.bind(this);
        this.eachNote = this.eachNote.bind(this);
    }

    componentWillMount() {
        const self = this;
        if (this.props.count) {
            $.getJSON("http://baconipsum.com/api/?type=all-meat&sentences=" + this.props.count + "&start-with-lorem=1&callback=?", function (results) {
                results[0].split('. ').forEach(function (sentence) {
                    self.add(sentence.substring(0, 40));
                });
            });
        }
    }

    nextId() {
        this.uniqueId = this.uniqueId || 0;
        return this.uniqueId++;
    }

    add(text) {
        const arr = this.state.notes;
        arr.push({
            id: this.nextId(),
            note: text
        });
        this.setState({notes: arr});
    }

    update(newText, i) {
        console.log("Updating item at index " + i + ", " + newText);
        const arr = this.state.notes;
        arr[i].note = newText;
        this.setState({notes: arr});
    }

    remove(i) {
        console.log("Removing item at index " + i);
        const arr = this.state.notes;
        const n = arr.splice(i, 1);
        this.setState({notes: arr});
    }

    eachNote(note, i) {
        return <Note key={note.id}
                     index={i}
                     onChange={this.update}
                     onRemove={this.remove}>{note.note}</Note>
    }

    render() {
        return (

            <div className="board">
                {this.state.notes.map(this.eachNote)}
                <button className="btn btn-sm btn-success glyphicon glyphicon-plus"
                        onClick={this.add.bind(null, "New Note")}/>
            </div>

        )
    }

}
export default Board;
