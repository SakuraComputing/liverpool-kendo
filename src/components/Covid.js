import React from 'react';

const Covid = () => (
    <div className="covid-content" >
        <h1>Coronavirus (COVID-19)</h1>
        <div className="covid-desc">Due to COVID the dojo will obviously remain closed until the government restrictions are lifted</div>
        <a 
            href="https://www.gov.uk/guidance/coronavirus-covid-19-grassroots-sports-guidance-for-safe-provision-including-team-sport-contact-combat-sport-and-organised-sport-events"
            className="covid-gov-link"
        >
            Government Contact Combat Sport Guidelines
        </a>
    </div>
);
export default Covid;