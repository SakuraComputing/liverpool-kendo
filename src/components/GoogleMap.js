import React from 'react';

import GoogleMapReact from 'google-map-react';

const AnyReactComponent = ({ text }) => <div>{text}</div>;

export class SimpleMap extends React.Component {
    static defaultProps = {
        center: {lat: 53.3701668, lng: -2.9195765},
        zoom: 15
    };

    render() {
        return (
            <GoogleMapReact
                bootstrapURLKeys={{ key: 'AIzaSyADAT64SMWOP7ee4cEOgW8tMYNL673MeDQ' }}
                defaultCenter={this.props.center}
                defaultZoom={this.props.zoom}
            >
                <AnyReactComponent
                    lat={53.3701668}
                    lng={-2.9195765}
                    text={'Liverpool Kendo Club'}
                />
            </GoogleMapReact>
        );
    }
}
export default SimpleMap;
