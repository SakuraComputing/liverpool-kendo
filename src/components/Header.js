import React from 'react';
import { Link } from 'react-router-dom';
import SideBar from './SideBar';

export const Header = () => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Liverpool Kendo Club</h1>
                </Link>
                <img src="/images/lkc_logo.jpg" alt="lkc_logo" className="header__icon" />
            </div>
        </div>
        <SideBar />
    </header>
);

export default Header;


