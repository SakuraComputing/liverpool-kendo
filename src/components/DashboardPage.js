import React from 'react';
import MainMenu from './SideBar';
import Address from './Address';

const DashboardPage = () =>(
    <div>
        <Address/>
    </div>
);
export default DashboardPage;
