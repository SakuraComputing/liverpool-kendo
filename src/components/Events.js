import React from 'react';
import Covid from './Covid';

export const Events = () => {
    return (
        <div className="content-container">
            <h1>Events coming up</h1>

            <h2>Seminars</h2>
            <Covid />

        </div>
    )
};
export default Events;
