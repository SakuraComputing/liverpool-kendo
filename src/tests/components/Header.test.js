import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Header, mapDispatchToProps } from '../../components/Header';

let dispatchStartLogoutProp, wrapper, currentUser;

describe('Header Page', () => {

    beforeEach(() => {
        dispatchStartLogoutProp = jest.fn();
        currentUser = jest.fn();
        wrapper = shallow(<Header dispatchStartLogoutProp={dispatchStartLogoutProp} currentUser={currentUser}/>);
    });

    it('should render header correctly',() => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('expect dispatchStartLogout to have been called on click', () => {
        wrapper.find('button').simulate('click');
        expect(dispatchStartLogoutProp).toHaveBeenCalled();
    });

    it('should map dispatch to props', () => {
        const mockDispatchStartLogout = jest.fn();
        const actionProps = mapDispatchToProps(mockDispatchStartLogout);
        actionProps.dispatchStartLogoutProp();
        expect(mockDispatchStartLogout).toHaveBeenCalledTimes(1);
    });
});


