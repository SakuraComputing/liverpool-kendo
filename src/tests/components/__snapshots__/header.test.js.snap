// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Liverpool Kendo Club
        </h1>
      </Link>
      <img
        alt="lkc_logo"
        className="header__icon"
        src="/images/lkc_logo.jpg"
      />
    </div>
  </div>
  <Connect(Sidebar) />
</header>
`;
